src/shared/telemetry/defaultTelemetryService.ts

```diff
@@ -278,37 +278,24 @@ export function filterTelemetryCacheEvents(input: any): MetricDatum[] {
             return true
         })
         .filter((item: Object) => {
-            // skip it if data is not an array or empty
-            if (!Array.isArray(item) || item.length === 0) {
-                getLogger().warn(`Item in telemetry cache: ${item}\n has invalid data field: ${item}! skipping!`)
+            // Only accept objects that have the required telemetry data
+            if (
+                !item.hasOwnProperty('Value') ||
+                !item.hasOwnProperty('MetricName') ||
+                !item.hasOwnProperty('EpochTimestamp') ||
+                !item.hasOwnProperty('Unit')
+            ) {
+                getLogger().warn(`Item in telemetry cache: ${JSON.stringify(item)}\n has invalid data! skipping!`)
```

Comment:
```suggestion
                getLogger().warn(`skipping invalid telemetry cache item: ${JSON.stringify(item)}\n`)
```

In general it is useful to put the variable part at the end of a log message. Especially in this case, where the variable part is multiline, having to scan the whole log to find the "skipping!" message at the end adds friction.